include/oneapi/dpl/internal/distributed_ranges_impl/concepts/concepts.hpp

```diff
@@ -0,0 +1,81 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <oneapi/dpl/internal/distributed_ranges_impl/detail/ranges.hpp>
```

Comment:
Is this header needed? If yes - why, and could not the necessary stuff be defined next to the concepts? It seems counter-intuitive that the definitions of concepts depend on something within the implementation.

Reply:
After experimenting to better understand how concepts are defined, I see why you need to have something from that header here. I will keep this comment open, but the action is on me to think of a preferred way to define the concepts while minimizing dependencies on the other API.

Reply:
I think that my confusion was partially caused by the file name and placement. You know, `detail/ranges.hpp` suggests that it has some internals of the DR implementation. I will make some suggestions in a new comment.